bundles/org.openhab.binding.juicenet/src/main/java/org/openhab/binding/juicenet/internal/handler/JuiceNetBridgeHandler.java

```diff
@@ -0,0 +1,204 @@
+
+        api.initialize(config.apiToken);
+
+        updateStatus(ThingStatus.UNKNOWN);
+        // Bridge will go online after the first successful API call in iterateApiDevices. iterateApiDevices will be
+        // called when a child device attempts to goOnline and needs to retrieve the api token
+
+        pollingJob = scheduler.scheduleWithFixedDelay(this::pollDevices, 10, config.refreshInterval, TimeUnit.SECONDS);
+
+        // Call here in order to discover any devices.
+        iterateApiDevices();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+        ScheduledFuture<?> pollingJob = this.pollingJob;
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            this.pollingJob = null;
+        }
+    }
+
+    public void setDiscoveryService(JuiceNetDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    @Override
+    public void childHandlerInitialized(ThingHandler childHandler, Thing childThing) {
+        // Call here to set the Api Token for any newly initialized Child devices
+        iterateApiDevices();
+    }
+
+    /**
+     * Get the services registered for this bridge. Provides the discovery service.
+     */
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(JuiceNetDiscoveryService.class);
+    }
+
+    public void handleApiException(Exception e) {
+        if (e instanceof JuiceNetApiException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.toString());
+        } else if (e instanceof InterruptedException) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, e.toString());
+            Thread.currentThread().interrupt();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.NONE, e.toString());
+        }
+    }
+
+    @Nullable
+    public Thing getThingById(String id) {
+        List<Thing> childThings = getThing().getThings();
+
+        for (Thing childThing : childThings) {
+            Configuration configuration = childThing.getConfiguration();
+
+            String childId = configuration.get(PARAMETER_UNIT_ID).toString();
+
+            if (childId.equals(id)) {
+                return childThing;
+            }
+        }
+
+        return null;
+    }
+
+    // This function will query the list of devices from the API and then set the name/token in the child handlers. If a
+    // child does not exist, it will notify the Discovery service. If it is successful, it will ensure the bridge status
+    // is updated
+    // to ONLINE.
+    public void iterateApiDevices() {
+        List<JuiceNetApiDevice> listDevices;
+        JuiceNetDiscoveryService discoveryService = this.discoveryService;
+
+        if (discoveryService == null) {
+            return;
```

Comment:
Shouldn't this method still set online status and notify child handlers, even if discovery service is not (yet) set?